### PR TITLE
[SPARK-45878][SQL][TESTS] Fix ConcurrentModificationException in CliSuite

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR changes the ArrayBuffer for logs to immutable for reading to prevent ConcurrentModificationException which hides the actual cause of failure

### Why are the changes needed?

```scala
[info] - SPARK-29022 Commands using SerDe provided in ADD JAR sql *** FAILED *** (11 seconds, 105 milliseconds)
[info]   java.util.ConcurrentModificationException: mutation occurred during iteration
[info]   at scala.collection.mutable.MutationTracker$.checkMutations(MutationTracker.scala:43)
[info]   at scala.collection.mutable.CheckedIndexedSeqView$CheckedIterator.hasNext(CheckedIndexedSeqView.scala:47)
[info]   at scala.collection.IterableOnceOps.addString(IterableOnce.scala:1247)
[info]   at scala.collection.IterableOnceOps.addString$(IterableOnce.scala:1241)
[info]   at scala.collection.AbstractIterable.addString(Iterable.scala:933)
[info]   at org.apache.spark.sql.hive.thriftserver.CliSuite.runCliWithin(CliSuite.scala:205)
[info]   at org.apache.spark.sql.hive.thriftserver.CliSuite.$anonfun$new$20(CliSuite.scala:501)
```

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala [hunk 1]
```diff
@@ -202,7 +202,7 @@ class CliSuite extends SparkFunSuite {
            |Exception: $cause
            |Failed to capture next expected output "${expectedAnswers(next)}" within $timeout.
            |
-           |${buffer.mkString("\n")}
+           |${buffer.toSeq.mkString("\n")}
```
Comment (LuciferYang) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala:205]: But the process of `.toSeq` still involves iteration, what is the difference between this iteration and the iteration of `.mkString`?

Comment (yaooqinn) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala:205]: You are right..

Comment (LuciferYang) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala:205]: Should we switch to a thread-safe collection like java.util.concurrent.ConcurrentLinkedQueue or use `synchronized`?

Comment (yaooqinn) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: I'd like to use the existing lock to synchronize both the buffer and next

**Comment by yaooqinn** [comment]
Thanks Merge to master

**Pull request closed without merging**